src/Cli/dotnet/commands/dotnet-workload/install/MsiInstallerBase.cs

```diff
@@ -497,9 +534,9 @@ protected void UpdateDependent(InstallRequestType requestType, string providerKe
         /// Deletes install state file for the specified feature band.
         /// </summary>
         /// <param name="sdkFeatureBand">The feature band of the install state file.</param>
-        protected void RemoveInstallStateFile(SdkFeatureBand sdkFeatureBand)
+        protected void RemoveInstallStateFile(string dotnetDir, SdkFeatureBand sdkFeatureBand)
```

Comment:
Great question!

How do we know if we're an MSI-based install or a file-based install? The current check goes to _dotnetDir_ to look for the msi sentinel file. That was the part that was actually throwing previously; as you said, we don't need to know dotnetDir if we're just writing to ProgramData, but we first check if we're an mis-based install, and that check threw an exception.

Reply:
This is in the MsiInstallerBase class, so we know we're in an MSI-based install already.  We do need to pass the `dotnetDir` to `WorkloadInstallType.GetInstallStateFolder`, but rather than passing it through these methods, I think we can use the `DotNetHome` property, and if not we should be able to figure it out some other way.